digdag-core/src/main/java/io/digdag/core/database/DatabaseSessionStoreManager.java

```diff
@@ -1550,6 +1554,9 @@ public StoredSessionAttempt getLastAttempt(long sessionId)
         public <T> T insertRootTask(long attemptId, Task task, SessionBuilderAction<T> func)
         {
             long taskId = dao.insertTask(attemptId, task.getParentId().orNull(), task.getTaskType().get(), task.getState().get(), task.getStateFlags().get());  // tasks table don't have unique index
+            // Root task's `started_at` isn't updated by trySetChildrenBlockedToReadyOrShortCircuitPlannedOrCanceled().
+            // So this line is just for setting `started_at` of root task although there may be room for optimization.
+            dao.setStartedState(taskId, task.getState().get(), task.getState().get());
```

Comment:
With this change `WorkflowExecutionTimeoutEnforcer.findTasksStartedBeforeWithState` will detect long running root and group tasks, but it makes `WorkflowExecutionTimeoutEnforcer` more sensitive. We need to make `WorkflowExecutionTimeoutEnforcer.findTasksStartedBeforeWithState` see only non-group tasks.